google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/BigtableClientContext.java

```diff
@@ -228,13 +230,13 @@ private static OpenTelemetry getOpenTelemetryFromMetricsProvider(
 
   private static void patchCredentials(EnhancedBigtableStubSettings.Builder settings)
       throws IOException {
-    int i = settings.getEndpoint().lastIndexOf(":");
-    String host = settings.getEndpoint().substring(0, i);
-    String audience = settings.getJwtAudienceMapping().get(host);
-
-    if (audience == null) {
-      return;
+    // Default jwt audience is always the service name unless it's override to
+    // test / staging for testing
+    String audience = DEFAULT_DATA_JWT_AUDIENCE;
+    if (settings.getJwtAudienceOverride() != null) {
+      audience = settings.getJwtAudienceOverride();
     }
```

Comment:
why not centralize this in stub settings?

class StubSetting.Builder {
  ctor {
     jwtAudience = "https://bigtable.googleapis.com/"
  }
  String getJwtAudience() { return jwtAudience }
  Builder setJwtAudience(s) { this.jwtAudience = s; return this; }
}

patchCredentials() {
  audience = settings.getJwtAUdience()
}
}